### PR TITLE
Add Y4 Frame type

### PR DESCRIPTION
**Pull request by Hwurzburg**
users requested.....flight tested
image
motororder-Y4a-2d
will also do a version via scripting as an example for the wiki for 2MB autopilots

### CODE REVIEW

File: libraries/AP_Motors/AP_MotorsMatrix.cpp [hunk 1]
```diff
@@ -1109,7 +1109,25 @@ void AP_MotorsMatrix::setup_motors(motor_frame_class frame_class, motor_frame_ty
...
+                // Y4 motor definition with right front CCW, left front CW
+                static const AP_MotorsMatrix::MotorDefRaw motors[] {
+                    { -1.0f,  1.000f, AP_MOTORS_MATRIX_YAW_FACTOR_CCW, 1 },
+                    {  0.0f, -1.000f, AP_MOTORS_MATRIX_YAW_FACTOR_CW,  2 },
+                    {  0.0f, -1.000f, AP_MOTORS_MATRIX_YAW_FACTOR_CCW, 3 },
+                    {  1.0f,  1.000f, AP_MOTORS_MATRIX_YAW_FACTOR_CW,  4 },
+                };
+                add_motors_raw(motors, ARRAY_SIZE(motors));
+            }
+                    break;
```
Comment (peterbarker) [inline libraries/AP_Motors/AP_MotorsMatrix.cpp:1129]: Slight indenting problem

**Comment by rmackay9** [comment]
As discussed, let's move this to the FRAME_TYPE.

**Comment by Hwurzburg** [comment]
Moved to frame type

File: libraries/AP_Motors/AP_MotorsMatrix.cpp [hunk 2]
```diff
@@ -765,6 +765,20 @@ void AP_MotorsMatrix::setup_motors(motor_frame_class frame_class, motor_frame_ty
                     add_motors(motors, ARRAY_SIZE(motors));
                     break;
                 }
+                case MOTOR_FRAME_TYPE_Y4:
+                    _mav_type = MAV_TYPE_QUADROTOR;
```
Comment (IamPete1) [inline libraries/AP_Motors/AP_MotorsMatrix.cpp:769]:
```suggestion
```
Don't need this it is set at the start of the quad case.

File: libraries/AP_Motors/AP_MotorsMatrix.cpp [hunk 3]
```diff
@@ -765,6 +765,20 @@ void AP_MotorsMatrix::setup_motors(motor_frame_class frame_class, motor_frame_ty
...
+                    _frame_type_string = "Y4";
+                    {
```
Comment (IamPete1) [inline libraries/AP_Motors/AP_MotorsMatrix.cpp:771]:
```suggestion

```
Don't think we need these brackets

File: libraries/AP_Motors/AP_MotorsMatrix.cpp [hunk 4]
```diff
@@ -765,6 +765,20 @@ void AP_MotorsMatrix::setup_motors(motor_frame_class frame_class, motor_frame_ty
...
+                    // Y4 motor definition with right front CCW, left front CW
+                    static const AP_MotorsMatrix::MotorDefRaw motors[] {
+                        { -1.0f,  1.000f, AP_MOTORS_MATRIX_YAW_FACTOR_CCW, 1 },
+                        {  0.0f, -1.000f, AP_MOTORS_MATRIX_YAW_FACTOR_CW,  2 },
+                        {  0.0f, -1.000f, AP_MOTORS_MATRIX_YAW_FACTOR_CCW, 3 },
+                        {  1.0f,  1.000f, AP_MOTORS_MATRIX_YAW_FACTOR_CW,  4 },
+                    };
+                    add_motors_raw(motors, ARRAY_SIZE(motors));
+                }
```
Comment (IamPete1) [inline libraries/AP_Motors/AP_MotorsMatrix.cpp:780]:
```suggestion
```

**Review by IamPete1 — CHANGES_REQUESTED** [review]

File: libraries/AP_Motors/AP_MotorsMatrix.cpp (hunk 2 above)
Comment (Hwurzburg) [reply libraries/AP_Motors/AP_MotorsMatrix.cpp:769]: removed...thanks

File: libraries/AP_Motors/AP_MotorsMatrix.cpp (hunk 3 above)
Comment (Hwurzburg) [reply libraries/AP_Motors/AP_MotorsMatrix.cpp]: removed

File: libraries/AP_Motors/AP_MotorsMatrix.cpp (hunk 4 above)
Comment (Hwurzburg) [reply libraries/AP_Motors/AP_MotorsMatrix.cpp]: removed

**Review by IamPete1 — APPROVED** [review]
LGTM

**Comment by rmackay9** [comment]
As discussed on the dev call, Y6 should really have been implemented as a type of hexacopter.  If we'd done that then we might have been able to share the "Y" between the "Y6" and this new "Y4".  Too late now though so approved!

**Review by rmackay9 — APPROVED** [review]

**Pull request merged**